src/finufft.cpp

```diff
@@ -92,6 +93,10 @@ Design notes for guru interface implementation:
 namespace finufft {
   namespace common {
 
+  // Technically global state...
```

Comment:
can you elaborate in this comment - are there two global states (one for single, one for double)? Is that ok?

Reply:
There's one lock since we marked it `static` -- it will use the same address if included in multiple compilation units. An alternative would be to make it a unique variable name for each compilation unit using a macro. (like `fftw_lock` and `fftwf_lock`). Technically, if you were running the single and double libraries at the same time you might desire having two distinct locks, but that's such a bizarre edge case it's probably not worth the ugliness.